### DISCUSSION TITLE
Content-Type not acceptable: application/vnd.api+json

### DISCUSSION THREAD

**Discussion by chrisallenmoore** [supabase/supabase#5661] [Questions] [answered]
I'm using orbit.js to sync data to Supabase. and I get a 400 error that says "Content-Type not acceptable: application/vnd.api+json".

Is this a content type that Supabase doesn't accept? Is it possible to turn it on?

```
const remote = new JSONAPISource({
      schema,
      name: 'remote',
      host: '{supabaseURL}',
      defaultFetchSettings: {
          headers: {
              'Content-Type': 'application/vnd.api+json',
              apikey: apikeydata,
              Authorization: authdata
          }
      }
  });
```

**Comment by steve-chavez** [comment, marked as answer]
> application/vnd.api+json

That's the [JSON API](https://jsonapi.org/) media type, which service(endpoint) are you trying to hit with that?

> Is this a content type that Supabase doesn't accept? Is it possible to turn it on?

AFAICT none of our services support JSON API, can you use a different method for your data source?

**Reply by chrisallenmoore** [reply]
@steve-chavez , that's correct it's the JSON API media type. I'm using Orbit.js and it uses that media type, but I've got the option to use `application/json` and it has gotten me a step further, and I am beginning to think it won't support it. After changing Accept and Content-Type to:
```
Accept: 'application/json',
'Content-Type': 'application/json',
```
I get the response error of 400 with the following message:
```
{message: "column "data" of relation "notes" does not exist", code: "42703", details: null, hint: null}
```
This is leading me to think that JSON API is indeed not supported.

I found a few references to this PG error code: 42703. But, I'm not sure.

use of syntactic sugar in treating it as a column:
https://gitter.im/begriffs/postgrest?at=56a117825de13b3f15e46ef8

prefixing your column names in the where clause:
https://stackoverflow.com/questions/17104837/pgsql-error-code-42703-column-does-not-exist

created your tables with double quotes, and now the names are case sensitive:
https://dba.stackexchange.com/questions/192897/postgres-relation-does-not-exist-error

Jackson ObjectConverter downcases the first letter:
https://githubmemory.com/repo/supabase-community/postgrest-kt/issues/4

I'm connecting with:

```
const remote = new JSONAPISource({
	schema,
	name: 'remote',
	host: 'https://<project>.supabase.co/rest/v1',
	defaultFetchSettings: {
		headers: {
			Accept: 'application/json',
			'Content-Type': 'application/json',
			apikey: apikeydata,
			Authorization: authdata
		}
	}
});
```

**Reply by steve-chavez** [reply]
> {message: "column "data" of relation "notes" does not exist", code: "42703", details: null, hint: null}

That error comes from PostgreSQL, and it means that the column `data` of the `notes` table is being requested - and data doesn't exist in this case. Somehow orbit(or JSON API) is requesting that.

> This is leading me to think that JSON API is indeed not supported.

Yes, as mentioned above. This would be better reported in https://github.com/PostgREST/postgrest/issues.

**Reply by chrisallenmoore** [reply]
@steve-chavez , thanks! I will do that. It must be how the JSON-API functions. Otherwise, I would have to model every part of the JSON-API schema. I appreciate your help!
